src/cmds/net/arping.c

```diff
@@ -114,15 +115,38 @@ int main(int argc, char **argv) {
 	from.s_addr = in_dev->ifa_address;
 	strncpy(from_b, inet_ntoa(from), sizeof(from_b) - 1);
 	printf("ARPING %s from %s %s\n", dst_b, from_b, in_dev->dev->name);
+
 	for (i = 1; i <= cnt; i++) {
 		neighbour_del(ETH_P_IP, &dst, in_dev->dev);
+
+		if (gettimeofday(&t1, NULL) == -1) {
+			return -1;
+		}
+
 		send_request(in_dev->dev, ETH_P_IP, sizeof in_dev->ifa_address,
 				&in_dev->ifa_address, &dst.s_addr);
-		usleep(DEFAULT_INTERVAL);
-		if (0 == neighbour_get_haddr(ETH_P_IP, &dst, in_dev->dev,
-					ARP_HRD_ETHERNET, sizeof hw_addr, &hw_addr[0])) {
+
+		/* Repeat until neigbour_get_haddr returns 0 or one second
+		 * (default interval) has passed.
+		 */
+		do {
+			ret = neighbour_get_haddr(ETH_P_IP, &dst, in_dev->dev,
+								ARP_HRD_ETHERNET, sizeof hw_addr, &hw_addr[0]);
+			if (gettimeofday(&t2, NULL) == -1) {
+				return -1;
+			}
+		} while (ret != 0 && (t2.tv_sec - t1.tv_sec) < 1);
```

Comment:
There is a possible situation when `(t2.tv_sec - t1.tv_sec) < 1` will happen already after a few microseconds. Let's image that `t1` is:

```
t1 = {
   .tv_sec = 0,
   .tv_usec = 999999
}
```

Then what will happen after one microsecond passed? Yes, you will get

```
t2 = {
    .tv_sec = 1,
    .tv_usec = 0
}
```

And the condition becomes true. So it's much better to try to use `timersub` here.

Reply:
Oh I see.. would something like this work?
```
            timersub(&t2, &t1, &sub_res);
        } while (ret != 0 && sub_res.tv_sec < 1);

        if (ret == 0 && sub_res.tv_sec < 1){
	    macaddr_print(mac, hw_addr);

	    /* To avoid doing floating point arithmetic. */
            microseconds = sub_res.tv_usec;
            (...)
```

Reply:
Why not? You can get a try :)

Reply:
I can't manage to push the change, when I do "git push origin 1723-arping-microseconds" I get the following error message:
```
 ! [rejected]            1723-arping-microseconds -> 1723-arping-microseconds (non-fast-forward)
error: failed to push some refs to 'https://github.com/eboswell/embox'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
```

Reply:
> "git push origin 1723-arping-microseconds"
Is it appears after rebase?
If so, you should 
"git push origin +1723-arping-microseconds"
But be careful. Check your git log before.